lib/benchee/profile.ex

```diff
@@ -0,0 +1,63 @@
+defmodule Benchee.Profile do
+  @moduledoc """
+  Profiles each scenario after benchmarking them if the `profile_after` flag is set to `true`.
+
+  The profiler that will be used is either the one set by the `profiler` option
+  in the `profile` configuration or the default one (`:cprof`). It accepts however the following
+  profiler options:
+    * `:cprof` will profile with `Mix.Task.Profile.Cprof` (useful to discover bottlenecks
+    related to function calls).
+    * `:eprof` will profile with `Mix.Task.Profile.Eprof` (useful to discover bottlenecks
+    related to time information of each function call).
+    * `:fprof` will profile with `Mix.Task.Profile.Fprof` (useful to discover bottlenecks
+    of a sequential code).
+  """
+  alias Benchee.Suite
+
+  @doc """
+  Runs for each scenario found in the suite the `profile/2` function from the given profiler.
+  """
+  @spec profile(Suite.t()) :: Suite.t()
+  def profile(suite = %{configuration: %{profile: %{profile_after: false}}}), do: suite
+
+  def profile(
+        suite = %{
+          configuration: %{profile: %{profiler: profiler, profiler_opts: profiler_opts}},
+          scenarios: scenarios
+        }
+      ) do
+    profiler_module = profiler_to_module(profiler)
+
+    Enum.each(scenarios, &run(&1, {profiler, profiler_module, profiler_opts}))
```

Comment:
personally not a fan of `&1` I know it's kinda apparent that it's `scenario` here but I'd rather have `fn -> .. end` here :)